### PR TITLE
DOC: Fix module links in module descriptions

### PR DESCRIPTION
**Pull request by jhlegarreta**
- DOC: Fix module names in module description ref links 
- DOC: Link groups in Doxygen in module description

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]

File: Modules/Filtering/AnisotropicSmoothing/itk-module.cmake [hunk 1]
```diff
@@ -3,7 +3,7 @@ anisotropic smoothing. This is an image denoising technique that strives for
 preserving edges on the images while smoothing regions of uniform intensity.
 This type of filtering is convenient as a preprocessing stage of segmentation
 algorithms. You may find useful as well the filters in the
-\\\\ref ITKCurvatureFlow module and the \\\\ref TKSmoothingModule.")
+\\\\ref ITKCurvatureFlow module and the \\\\ref ITKSmoothing.")
```
Comment (N-Dekker) [inline Modules/Filtering/AnisotropicSmoothing/itk-module.cmake:6]: Perhaps add the word "module"here, as in `the \\\\ref ITKSmoothing module` ?

File: Modules/Filtering/CurvatureFlow/itk-module.cmake [hunk 2]
```diff
@@ -2,7 +2,8 @@ set(DOCUMENTATION "This module contains filters that implement variations of
 Curvature Flow. This is a technique that uses an iterative solution of partial
 differential equations to implement image denoising image filtering. These
 classes are typically used as edge-preserving smoothing filters. You may also
-find the ITKSmoothing and the ITKAnisotropicSmoothing useful as well.")
+find the \\\\ref ITKSmoothing and the \\\\ref ITKAnisotropicSmoothing useful
```
Comment (N-Dekker) [inline Modules/Filtering/CurvatureFlow/itk-module.cmake:5]: Perhaps add the word "module" here, as in `the \\\\ref ITKSmoothing and the \\\\ref ITKAnisotropicSmoothing module` ?

File: Modules/Filtering/GPUAnisotropicSmoothing/itk-module.cmake [hunk 3]
```diff
@@ -3,7 +3,7 @@ implement variations of anisotropic smoothing. This is an image denoising
 technique that strives for preserving edges on the images while smoothing regions
 of uniform intensity.  This type of filtering is convenient as a preprocessing
 stage of segmentation algorithms. You may find useful as well the filters in the
-\\\\ref ITKGPUSmoothingModule.")
+\\\\ref ITKGPUSmoothing.")
```
Comment (N-Dekker) [inline Modules/Filtering/GPUAnisotropicSmoothing/itk-module.cmake:6]: Perhaps the `\\\\ref ITKGPUSmoothing module` ?

File: Modules/Filtering/AnisotropicSmoothing/itk-module.cmake (hunk 1 above)
Comment (jhlegarreta) [reply Modules/Filtering/AnisotropicSmoothing/itk-module.cmake]: ITK's scripts automatically add a `Module` prefix to these, so no need for that. Please have a look at the Doxygen files. In fact, I removed the redundant `module` wording in here in 5cb6105.

File: Modules/Filtering/CurvatureFlow/itk-module.cmake (hunk 2 above)
Comment (jhlegarreta) [reply Modules/Filtering/CurvatureFlow/itk-module.cmake:5]: ITK's scripts automatically add a `Module` prefix to these, so no need for that. Please have a look at the Doxygen files.

File: Modules/Filtering/GPUAnisotropicSmoothing/itk-module.cmake (hunk 3 above)
Comment (jhlegarreta) [reply Modules/Filtering/GPUAnisotropicSmoothing/itk-module.cmake:6]: ITK's scripts automatically add a `Module` prefix to these, so no need for that. Please have a look at the Doxygen files.

**Review by N-Dekker — APPROVED** [review]
Thanks, Jon!

**Comment by jhlegarreta** [comment]
Related doxygen warnings are fixed (the unbalanced warnings to address at some point) and links are working now.

**Pull request merged**